source/docs/software/networktables/creating-sendable-classes.rst

```diff
@@ -0,0 +1,117 @@
+
+What is SendableBuilder?
+------------------------
+
+SendableBuilder (`Java <https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/smartdashboard/SendableBuilder.html>`__, `C++ <https://first.wpi.edu/FRC/roborio/release/docs/cpp/classfrc_1_1SendableBuilder.html>`__), implemented in SendableBuilderImpl, handles NetworkTable interaction with an instance of a Sendable class. Through the SendableBuilder instance passed in the :code:`initSendable` method of the Sendable interface, Sendable subclasses can quickly publish the relevant information to NetworkTables. The :code:`initSendable` method is called when a user puts the Sendable on SmartDashboard or Shuffleboard, by calling the code shown above. Users may add properties to the SendableBuilder passed into the :code:`initSendable` method, much like the old SmartDashboard methods used to put primatives onto SmartDashboard. The full list of properties which can be published to and retrieved from SmartDashboard can be found on the linked SendableBuilder documentation, and includes:
+
+- Boolean values
+- Boolean arrays
+- Double values
+- Double arrays
+- Bytes values
+- String values
+- String arrays
+- NetworkTableValues
+
+SendableBuilder can also be used to set the actuator flag on or off, add functions to run to set the Sendable into a safe state, set the type of the Sendable displayed on SmartDashboard, or functions to run to update the network table for things other than properties. If a Sendable is configured as an actuator, users will be able to control it directly through Shuffleboard, SmartDashboard or OutlineViewer when the robot is enabled in Test mode.
+
+Creating a new Sendable class with SendableBuilder
+--------------------------------------------------
+
+To expose information about a class to users over NetworkTables, one must create an implementation of the initSendable() method of the Sendable interface. The properties added to Sendable classes can include just a getter, or both a getter and a setter. This example shows a Double property which can both be viewed and modified on NetworkTables. Note that both a getter and setter are not required - a :code:`null` setter simply means the value can't be changed over NetworkTables. This might be desired in some situations where information should only be displayed, but not modified. If the Sendable is configured as an actuator, the setter will be called when the robot is enabled in Test mode when users change its values to test robot functions.
+
+.. tabs::
+
+    .. code-tab:: c++
+
+        void DifferentialDrive::InitSendable(SendableBuilder& builder) {
```

Comment:
This example clearly is not for a differential drive class.

An example that uses all the methods explained the section header would be appropriate here. A single property (especially one from something that is itself `Sendable`) isn't very useful.

Also needs to use `SetSmartDashboardType` as that's what lets dashboards use custom widgets for display and control. Otherwise the sendable could only be displayed as a basic list of properties